ckcp/run.sh

```diff
@@ -58,3 +58,91 @@ KUBECONFIG=kubeconfig/admin.kubeconfig kubectl api-resources
 #test the registration of a Physical Cluster
 curl https://raw.githubusercontent.com/kcp-dev/kcp/main/contrib/examples/cluster.yaml > cluster.yaml
 sed -e 's/^/    /' $KUBECONFIG | cat cluster.yaml - | KUBECONFIG=kubeconfig/admin.kubeconfig kubectl apply -f -
+
+
+#install pipelines/triggers based on args
+if [ $# -eq 0 ]; then
+  echo "No args passed; exiting now! ckcp is running in a pod"
+else
+  for arg in "$@"
+  do
+    if [ $arg == "pipelines" ]; then
+      echo "Arg $arg passed. Installing pipelines in ckcp"
+      WORKING_DIR=work/
+      mkdir -p $WORKING_DIR
+      cd $WORKING_DIR
+
+      if [[ ! -d ./pipeline ]]
+      then
+        git clone git@github.com:tektoncd/pipeline.git
+        (cd ./pipeline && git checkout v0.32.0)
+
+        # Pods need to be placed on a physical cluster
+        # Adding the label manually for that purpose.
+        (cd pipeline && git apply ../../../label.patch)
+
+        # This feature unblocks Tekton pods. The READY annotation is not correctly propagated to the physical cluster.
+        (cd pipeline && git apply ../../../pipeline-ff.patch)
+
+        # Conversion is not working yet on KCP
+        (cd pipeline && git apply ../../../remove-conversion.patch)
+      fi
+
+      if [[ ! -f ./pipeline/bin/controller ]]
+      then
+        (cd ./pipeline && make bin/controller)
+      fi
+
+      #install namespaces in ckcp
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl create namespace default
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl create namespace tekton-pipelines
+
+      #install pipelines CRDs in ckcp
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply -f pipeline/config/300-pipelinerun.yaml
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply -f pipeline/config/300-taskrun.yaml
+
+      # will go away with v1 graduation
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply -f pipeline/config/300-run.yaml
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply -f pipeline/config/300-resource.yaml
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply -f pipeline/config/300-condition.yaml
+
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl apply $(ls pipeline/config/config-* | awk ' { print " -f " $1 } ')
+
+      # Test 3 - create taskrun and pipelinerun
+
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl create serviceaccount default
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl create -f pipeline/examples/v1beta1/taskruns/custom-env.yaml
+      KUBECONFIG=../kubeconfig/admin.kubeconfig kubectl create -f pipeline/examples/v1beta1/pipelineruns/using_context_variables.yaml
+
+      KUBECONFIG=../kubeconfig/admin.kubeconfig METRICS_DOMAIN=knative.dev/some-repository SYSTEM_NAMESPACE=tekton-pipelines KO_DATA_PATH=./pipeline/pkg/pod/testdata ./pipeline/bin/controller \
```

Comment:
- `METRICS_DOMAIN` could point to something like `openshift.io/pipeline-service` or something (even though not really useful today).
- Flags might evolve over time upstream, right now it got "extracted" manually right ? I wonder if we could have a small "script" that extract the command/args from the release.yaml.

Reply:
Thanks for this idea. I like extracting info from latest release.yaml and using it here. I will work on this and raise another PR. For now, have raised this PR to accommodate running pipelines.

https://github.com/openshift-pipelines/pipelines-service/pull/18